tests/end_to_end.rs

```diff
@@ -83,7 +83,7 @@ fn test_cleanup() {
         })
         .collect();
 
-    let expected_variables = HashSet::from(["HOME", "_", "PWD", "SHLVL"]);
```

Comment:
why does HOME go away from this list?

Reply:
It was only being included in the list due to the `unwrap_or_default()` present before. HOME is not being included even if we don't run `command.env_clear()`, I'm not sure why 🤔 

Maybe it's a problem with my machine? HOME wasn't present even before this PR

Reply:
well, aren't we expecting HOME to get set here? Maybe Cargo clears it out or something?

Reply:
I'm trying to discover this right now. Cargo doesn't seem to the culprit here, running `env::var("HOME")` in another project captures `HOME` properly. Nix also doesn't seem to be the problem, as running `nix-shell` and then running `echo $HOME` also prints it properly.
I can investigate the issue, but since we have plans to set a fake `HOME`, I was wondering if we could leave it as is?

Reply:
Weird. I’ll have a try for myself later on!

Reply:
yeah, looks like the whole environment is getting cleared somewhere earlier. I'm gonna just remove the special HOME handling and merge this. We're close enough to creating a HOME that this is ignoreable.